filters/apiusagemonitoring/spec.go

```diff
@@ -290,58 +286,61 @@ func (s *apiUsageMonitoringSpec) buildClientTrackingInfo(apiIndex int, api *apiC
+	rexexpSlashColumnVar    = regexp.MustCompile(`/:([^:{}/]*)`)
+	rexexpCurlyBracketVar   = regexp.MustCompile(`{([^{}]*?)([?]?)}`)
+	regexpEscapeBeforeChars = regexp.MustCompile(`([.*+?\\])`)
+	regexpEscapeAfterChars  = regexp.MustCompile(`([{}[\]()|])`)
+)
 
-	matcher = rawRegEx.String()
-	normalizedPathTemplate = strings.Join(normalizedPathTemplateParts, "/")
-	pathLabel = strings.Join(pathLabelParts, "/")
+// PathHandler path handler interface.
+type PathHandler interface {
+	NormalizePathTemplate(path string) string
+	CreatePathPattern(path string) string
+}
 
-	return
+// defaultPathHandler default path handler implementation.
+type defaultPathHandler struct{}
+
+// NormalizePathTemplate normalize path template removing the leading and
+// trailing slashes, substituting multiple adjacent slashes with a single
+// one, and replacing column based variable declarations by curly bracked
+// based.
+func (ph defaultPathHandler) NormalizePathTemplate(path string) string {
+	path = regexpLeadingSlashes.ReplaceAllString(path, "")
+	path = regexpTrailingSlashes.ReplaceAllString(path, "")
+	path = regexpMultipleSlashes.ReplaceAllString(path, "/")
+	path = rexexpSlashColumnVar.ReplaceAllString(path, "/{$1}")
+	path = rexexpCurlyBracketVar.ReplaceAllString(path, "{$1}")
+	return path
 }
 
-// getVarPathPartName detects if a path template part represents a variadic placeholder.
-// Returns "" when it is not or its name when it is.
-func getVarPathPartName(pathTemplatePart string) string {
-	// check if it is `:id`
-	matches := regexVarPathPartColon.FindStringSubmatch(pathTemplatePart)
-	if len(matches) == 2 {
-		return matches[1]
-	}
-	// check if it is `{id}`
-	matches = regexVarPathPartCurlyBraces.FindStringSubmatch(pathTemplatePart)
-	if len(matches) == 2 {
-		return matches[1]
+// CreatePathPattern create a regular expression path pattern for a path
+// template by escaping regular specific characters, add optional matching
+// of leading and trailing slashes, accept adjacent slashes as if a single
+// slash was given, and allow free matching of content on variable locations.
+func (ph defaultPathHandler) CreatePathPattern(path string) string {
+	path = regexpEscapeBeforeChars.ReplaceAllString(path, "\\$1")
+	path = rexexpSlashColumnVar.ReplaceAllString(path, "/.+")
+	path = rexexpCurlyBracketVar.ReplaceAllStringFunc(path, ph.SelectPathVarPattern)
+	path = regexpLeadingSlashes.ReplaceAllString(path, "^/*")
+	path = regexpTrailingSlashes.ReplaceAllString(path, "/*$")
+	path = regexpMiddleSlashes.ReplaceAllString(path, "$1/+$2")
+	path = regexpEscapeAfterChars.ReplaceAllString(path, "\\$1")
+	return path
+}
+
+// selectPathVarPattern select the correct path variable pattern depending
+// on the path variable syntax. A trailing question mark is interpreted as
+// a path variable that is allowed to be empty.
+func (ph defaultPathHandler) SelectPathVarPattern(match string) string {
+	log.Infof("match %s", match)
```

Comment:
I think this log entry should be enabled only in debug mode. Please change it to Debugf.